framework/include/utils/FileLineInfo.h

```diff
@@ -0,0 +1,120 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef FILELINEINFO_H
+#define FILELINEINFO_H
+
+#include <string>
+#include <map>
+
+/**
+ * Holds file and line information.
+ */
+class FileLineInfo
+{
+public:
+  FileLineInfo();
+  FileLineInfo(const std::string & f, int l);
+  /**
+   * @return Whether this holds valid file line information.
+   */
+  bool isValid() const;
+  int line() const;
+  std::string file() const;
+
+protected:
+  int _line;
+  std::string _file;
+};
+
+/**
+ * A mapping between a series of keys to a FileLineInfo.
+ * This is intended to replace having a std::pair or a std::tuple
+ * as a key to a map holding the FileLineInfo.
+ */
+class FileLineInfoMap
+{
+public:
+  /**
+   * Associate a key with file/line info
+   * @param key0 Key
+   * @param file file
+   * @param line line number
+   */
+  void addInfo(const std::string & key0, const std::string & file, int line);
```

Comment:
Is it possible to use C++11 variadic arguments for the keys? Just to avoid having three definitions for the add and get methods.